timesketch/lib/analyzers/sigma_tagger.py

```diff
@@ -35,11 +35,22 @@ def run_sigma_rule(self, query, rule_name, tag_list = None):
         """
         return_fields = []
         tagged_events_counter = 0
+        # TODO: tune the below line to be more effective
         events = self.event_stream(
             query_string=query, return_fields=return_fields)
         for event in events:
-            event.add_tags(['sigma_{0:s}'.format(rule_name)])
-            event.add_tags(tag_list)
+            ts_sigma_rule = event.source.get('ts_sigma_rule',[])
+            ts_sigma_rule.append(rule_name)
+            event.add_attributes({'ts_sigma_rule': list(set(ts_sigma_rule))})
+            ts_ttp = event.source.get('ts_ttp',[])
+            for tag in tag_list:
+                print(tag)
+                if tag.startswith(('attack.','car.')):
+                    ts_ttp.append(tag)
+                else:
+                    event.add_tags([tag])
```

Comment:
why do this, that is for each tag you call the `add_tags` function, instead of just calling it once with the list of tags you want to add, that way you can save yourself multiple API Calls

Reply:
Good point, hope my new code is clever enough, I basically remove elements that will be added to the new attribute from the taglist and then take what is left.